pkg/helm/actions/get_chart.go

```diff
@@ -1,19 +1,238 @@
 package actions
 
 import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/openshift/console/pkg/helm/chartproxy"
 	"helm.sh/helm/v3/pkg/action"
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/chart/loader"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/dynamic"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
-func GetChart(url string, conf *action.Configuration) (*chart.Chart, error) {
+// constants
+const (
+	configNamespace         = "openshift-config"
+	tlsSecretCertKey        = "tls.crt"
+	tlsSecretKey            = "tls.key"
+	caBundleKey             = "ca-bundle.crt"
+	tlsSecretPattern        = "tlscrt-*"
+	tlsKeyPattern           = "tlskey-*"
+	cacertPattern           = "cacert-*"
+	openshiftRepoUrl        = "https://charts.openshift.io"
+	openshiftChartUrlPrefix = "https://github.com/openshift-helm-charts/"
+)
 
-	cmd := action.NewInstall(conf)
+func writeTempFile(data []byte, pattern string) (*os.File, error) {
+	f, createErr := os.CreateTemp("", pattern)
+	if createErr != nil {
+		return nil, createErr
+	}
+
+	_, writeErr := f.Write(data)
+	if writeErr != nil {
+		return nil, writeErr
+	}
+
+	closeErr := f.Close()
+	if closeErr != nil {
+		return nil, closeErr
+	}
+
+	return f, nil
+}
+
+func getChartNameAndNamespaceFromChartUrl(url, namespace string, client dynamic.Interface, coreClient corev1client.CoreV1Interface) (string, string, error) {
+	// create an index.yaml from all repository
+	//iterate all entries and find chart with the right url
+	var repositoryName, repositoryNamespace string
+	// indexFile, err := proxy.IndexFileAll(true)
+	// if err != nil {
+	// 	return "", "", fmt.Errorf("Error In Finding the chart repositories")
+	// }
+	// for _, entry := range indexFile.Entries {
+	// 	for i := len(entry) - 1; i >= 0; i-- {
+	// 		if containsUrl(url, entry[i].URLs) {
+	// 			return entry[i].Annotations["repositoryName"], entry[i].Annotations["repositoryNamespace"], nil
+	// 		}
+	// 	}
+	// }
+	repoGetter := chartproxy.NewRepoGetter(client, coreClient)
+	helmRepo, err := repoGetter.List(namespace)
+	//helmRepo, err := FindAllRepositories(client, coreClient, namespace)
+	if err != nil {
+		return "", "", fmt.Errorf("Error In Finding the chart repositories")
+	}
+	for _, repo := range helmRepo {
+		if strings.HasPrefix(url, repo.URL.String()) {
+			return repo.Name, repo.Namespace, nil
+		} else if repo.URL.String() == openshiftRepoUrl && strings.HasPrefix(url, openshiftChartUrlPrefix) {
+			return repo.Name, repo.Namespace, nil
+		}
+		//repositoryName, repositoryNamespace = FindChartWithUrl(url, idx)
+		// if repositoryName != "" || repositoryNamespace != "" {
+		// 	return repositoryName, repositoryNamespace, nil
+		// }
+	}
+	return repositoryName, repositoryNamespace, fmt.Errorf("Not Found")
```

Comment:
`repositoryName` and `repositoryNamespace` might be removed and empty strings used here.